modules/Kconfig.mcuboot

```diff
@@ -1,6 +1,17 @@
 config SIGN_IMAGES
 	bool "Sign images for MCUBoot"
-	default y
+	default y if MCUBOOT_BUILD_STRATEGY_FROM_SOURCE
 	help
 	  Sign images for MCUBoot as integrated part of the build stages using
 	  the private key.
+
+config BOOT_SIGNATURE_KEY_FILE
+	string "MCUBoot PEM key file (REQUIRED)"
```

Comment:
```suggestion
 	string "MCUBoot PEM key file" 
```
Have not seen such in Kconfig before, although there are settings that requires a value.

Actually, thinking about this, we could turn the dependency around, and disable signing if `BOOT_SIGNATURE_KEY_FILE` is not set.

like:
```
config SIGN_IMAGES
	bool "Sign images for MCUBoot"
	depends on MCUBOOT_BUILD_STRATEGY_FROM_SOURCE || BOOT_SIGNATURE_KEY_FILE != ""
	default y
	help
	  Sign images for MCUBoot as integrated part of the build stages using
	  the private key.
```

This means this setting is only available if user is building from source or have specified the pem key.

And thus I believe the CMake checking can go away.